### PR TITLE
Update idea template files

### PR DESCRIPTION
**Pull request by stevemessick**
The IntelliJ-specific template files used to create Flutter projects need to be cleaned up.

- Moves the android module file into the android directory for apps and plugins. That was requested but not considered essential. It is the easiest way to get language-specific source files into the module file. There is no comparable change for the module template since they currently do not support kotlin, and any changes there would be fragile. The package template did not need any changes.

- Adds a new module for plugins. Formerly, the android module had two content roots, one for the module and another for the android module in the example app. Now the example app has its own android module.

- Adds a library definition for KotlinJavaRuntime, so users don't have to manually do it. (The .idea/libraries file is also included in Java projects but is harmless; the library is not defined. It is difficult to avoid generating it since the .idea directory does not support multiple languages.)

- GeneratedPluginRegistrant.java is generated for kotlin apps. The app template needs both java and kotlin src directories in its module file.

Aside: Plugins have yellow background in the Project pane due to the plugin directory being defined as a library root in `.idea/libraries/Flutter_Plugins.xml`. This can be changed in `Preferences > Appearance & Behavior > File Colors > Non-Project Files`.

@devoncarew @pq @jwren

### CODE REVIEW

**Comment by devoncarew** [comment]
Looks like there's a failure in https://cirrus-ci.com/task/4726752750338048, in the `test/commands/create_test.dart: kotlin/swift plugin project` test.

**Review by devoncarew — COMMENTED** [review]
The changes look good here; will re-review once the build is green.

cc @matthew-carroll in case the change to the IntelliJ metadata for the android side of the app templates is interesting

**Comment by stevemessick** [comment]
@devoncarew The tests all pass now.

File: packages/flutter_tools/test/commands/create_test.dart [hunk 1]
```diff
@@ -278,7 +279,7 @@ void main() {
         'ios/Classes/FlutterProjectPlugin.h',
         'ios/Classes/FlutterProjectPlugin.m',
         'ios/Classes/SwiftFlutterProjectPlugin.swift',
-        'lib/flutter_project.dart',
+        'lib/flutter_project.dart'
```
Comment (devoncarew) [inline packages/flutter_tools/test/commands/create_test.dart]: nit: we likely still want the trailing comma

**Review by devoncarew — APPROVED** [review]

File: packages/flutter_tools/test/commands/create_test.dart (hunk 1 above)
Comment (stevemessick) [reply packages/flutter_tools/test/commands/create_test.dart]: Yes, that was a typo. I've restored it.

**Pull request merged**